### PR TITLE
Get all dependencies from the requirements.txt file

### PR DESCRIPTION
**Pull request by HanslettTheDev**
## Summary

This pull request solves this [Issue](https://github.com/PyAr/pyafipws/issues/105). The setup.py file has some other dependencies which setup.py has to be aware of so as to build the project. This PR solves that issue

## Checklist

- [X] Classes, Variables, function and methods logic  ok
- [X] Comments written explaining what the code does
- [X] All python code is PEP8 compliant (run black .)

### CODE REVIEW

**Review by reingart — CHANGES_REQUESTED** [review]
Hi @HanslettTheDev , sorry I missed this PR and I thought we already discussed this approach in #105

Just in case, here go my concerns / comments again:
* Dependencies should be explicitly declared in `install_requires`, with proper conditions for both python versions (changing `==` with `>=`  will not work for some dependencies like cryptography, minimal version with upper bound should be used)
* Instead of creating an ad-hoc custom script (get_dep.py), a generic one should be used, like [pip-tools](https://pypi.org/project/pip-tools/), so library versions could be resolved automatically in the future and manual pinning is avoided
* `requirements.txt` should not be used from `setup.py` (not a good practice), as it is the dependency freezing / locking for our tests but other projects that uses the library could need to use different library versions for compatibility

Python packaging documentation has more information about this topics and examples:
* https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#install-requires
* https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#install-requires-vs-requirements-files

**Comment by HanslettTheDev** [comment]
Below is an image showing pip-compile being executed

image

File: setup.py [hunk 1]
```diff
@@ -70,6 +70,23 @@
...
+        "cryptography>=3.4.7;python_version > '3'",
+        "fpdf>=1.7.2",
+        "dbf>=0.88.019",
+        "Pillow>=2.0.0",
+        "tabulate>=0.8.5",
+        "certifi>=2020.4.5.1",
+        "qrcode>=6.1",
+        "future>=0.18.2",
+        "pywin32==304;sys_platform == 'win32' and python_version > '3'",
+        "py2exe==0.11.1.1;sys_platform == 'win32' and python_version > '3'"
```
Comment (reingart) [inline setup.py]: py2exe should be a development dependency, not really needed for installation and normal operation

File: setup.py [hunk 2]
```diff
@@ -70,6 +70,23 @@
...
+        "cryptography==3.3.2;python_version <= '2.7'",
+        "cryptography>=3.4.7;python_version > '3'",
+        "fpdf>=1.7.2",
+        "dbf>=0.88.019",
+        "Pillow>=2.0.0",
+        "tabulate>=0.8.5",
+        "certifi>=2020.4.5.1",
+        "qrcode>=6.1",
+        "future>=0.18.2",
+        "pywin32==304;sys_platform == 'win32' and python_version > '3'",
```
Comment (reingart) [inline setup.py]: pywin32 should be an optional dependency, it is required to use this library from 3rd party languages (python apps don't need it)

**Review by reingart — CHANGES_REQUESTED** [review]
Thanks! I left a couple of comments, maninly about windows dependencies that should be optional or just for development (compiling the executable for 3rd party languages)

**Review by reingart — APPROVED** [review]

File: setup.py [hunk 3]
```diff
@@ -70,6 +70,24 @@
     author_email="reingart@gmail.com",
     url="https://github.com/reingart/pyafipws",
     license="LGPL-3.0-or-later",
+    install_requires=[
+        "httplib2==0.9.2;python_version <= '2.7'",
+        "httplib2>=0.20.4;python_version > '3'",
+        "pysimplesoap==1.08.14;python_version <= '2.7'",
+        "pysimplesoap @ git+https://github.com/pysimplesoap/pysimplesoap.git@py311#pysimplesoap;",
```
Comment (reingart) [inline setup.py]: Here it is missing the python 3 conditional, right?
```suggestion
        "pysimplesoap @ git+https://github.com/pysimplesoap/pysimplesoap.git@py311#pysimplesoap; python_version > '3'",
```

**Review by reingart — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (HanslettTheDev) [reply setup.py]: done

**Comment by reingart** [comment]
Hi @HanslettTheDev , it seems there is an error in the requirements format, can you check it?
```
error in PyAfipWs setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Parse error at ""> '3'"": Expected stringEnd
Error: Process completed with exit code 1.
```
Probably we should publish the pysimplesoap package and use it instead of the github repo

**Comment by HanslettTheDev** [comment]
> Hi @HanslettTheDev , it seems there is an error in the requirements format, can you check it?
> 
> ```
> error in PyAfipWs setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Parse error at ""> '3'"": Expected stringEnd
> Error: Process completed with exit code 1.
> ```
> 
> Probably we should publish the pysimplesoap package and use it instead of the github repo

I'll remove the markers for the pysimplesoap and then we merge. I'll open another ticket to solve the pysimplesoap just to be sure there are no extra dependencies that could fail along the way

**Review by reingart — APPROVED** [review]

**Pull request merged**